app/js/enhancedDate.js

```diff
@@ -1,5 +1,156 @@
-/* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+var enhancedDate = (function (date) {
+// still maintains access to all globals
+  
+  var dateMethods = {};
+  var myDate = date;
+  var dateString = Date(year, month, day, hours, minutes, seconds, milliseconds);
+  var newDate = new Date();
+  var epochDate = newDate.getTime();
+  
+//Stores the passed-in date.  
+  dateMethods.checkDate = function () {
```

Comment:
This sounds like setDate (but it's called checkDate?). This method should be setting a date, which, in the case of your module, should be myDate, but it looks like it just returns true or false instead. So in this method, which should take a parameter(dateMethods.setDate = function (date) { ) , you would be setting myDate = new Date( ) and pass in date if date is an instance of Date or if date is a number ( if (date instanceof Date || typeof date === 'number') { myDate = new Date(date); }